ApiReview/API.Polly.Core/NoDocs/Polly.Timeout/OnTimeoutArguments.cs

```diff
@@ -0,0 +1,14 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Polly.Timeout;
+
+public sealed class OnTimeoutArguments
+{
+    public ResilienceContext Context { get; }
+    public Exception Exception { get; }
```

Comment:
The arguments have this non-nullable Exception property which seem to imply that the timeout outcome is always a result of some code throwing an exception (OperationCanceledException I assume). did I get it right?

do we support the case when a delegate doesn't support cancellation via token?

Reply:
> The arguments have this non-nullable Exception property which seem to imply that the timeout outcome is always a result of some code throwing an exception (OperationCanceledException I assume). did I get it right? 

Yes, `OnTimeoutArguments` is only ever reported if the passed callback respected the cancellation token. I think we can drop `Exception`, I don't see it having much use and we can still add it in backward-compatible way later if there is a need for it.

Reply:
Drop it

Reply:
Done.